src/compiler.rkt

```diff
@@ -211,250 +208,390 @@
   (define (<compiled-prog> . xs) (vector-ref (apply core xs) 0))
   <compiled-prog>)
 
-(define (backward-pass ivec varc vregs vprecs vstart-precs root-reg)
+(define (backward-pass ivec varc vregs vprecs vstart-precs roots)
   (vector-fill! vprecs 0)
-
-  (define result (vector-ref vregs root-reg))
-  (when
-      (equal? 1 (flonums-between
-                 (bigfloat->flonum (ival-lo result))
-                 (bigfloat->flonum (ival-hi result))))
-    (vector-set! vprecs (- root-reg varc) (get-slack)))
+  (for ([root-reg (in-vector roots)])  ; adding slack in case rounding boundary
+    (when (and
+           (<= 0 (- root-reg varc))                                 ; when root is not a variable
+           (not (equal? (vector-ref ivec (- root-reg varc)) const)) ; when root is not a const
```

Comment:
I don't think this will work? Like, I don't think this ever fires?

Reply:
First of all `ivec[root-reg - varc]` will be a list whose first element is a constant function. Second of all, `const` is a function that returns constant functions, it's not equal to them. So I think you can just remove this conjunct.

Reply:
I see, agree